### ISSUE TITLE
[java] PublicMemberInNonPublicType: False positive for `main(...)` methods

### ISSUE THREAD

**Issue by uhafner** [pmd/pmd#6459]
**Affects PMD Version: 7.22.0** 

**Rule: [PublicMemberInNonPublicType](https://pmd.github.io/pmd/pmd_rules_java_design.html#publicmemberinnonpublictype)**

Please provide the rule name and a link to the rule documentation:

**Description:**

The new rule flags the usage of the `main` method in non public classes. I think only newer JDK version allow main methods to be non-public.

**Code Sample demonstrating the issue:**

```java
class MyTest {
    public static void main(final String... args)  {
        // main method that is invoked in IntelliJ
    }
}
```

**Expected outcome:**

PMD reports a violation, but that's wrong. That's a false positive.

**Running PMD through:** *Maven*

**Caused by:** #6231

**Comment by uhafner** [comment]
~~BTW: In order to ignore the violation I need to suppress on the type and not on the affected method, it would be nice if that can be changed as well.~~ (As shown in https://github.com/jenkinsci/analysis-model/blob/666ae125babdfb716e5c1c5198acf01559a7f008/src/test/java/edu/hm/hafner/analysis/IssueTest.java#L362 this works for methods)

**Comment by UncleOwen** [comment]
Is this really a false positive?

In Java < 25 (ignoring previews), this isn't a valid entry point.

In Java >= 25, you can write this as
```java
class MyTest {
    void main(final String... args)  {
        // main method that is invoked in IntelliJ
    }
}
```
or even
```java
void main(final String... args)  {
    // main method that is invoked in IntelliJ
}
```

In both cases, I'd say that the warning is justified?

**Comment by uhafner** [comment]
> Is this really a false positive?
> 
> In Java < 25 (ignoring previews), this isn't a valid entry point.

It is a valid entry point in Java < 25. Why should this not be possible? 

I am using this part in a lot of my test cases, example: https://github.com/jenkinsci/analysis-model/blob/666ae125babdfb716e5c1c5198acf01559a7f008/src/test/java/edu/hm/hafner/analysis/IssueTest.java#L362

**Comment by UncleOwen** [comment]
Huh. I could have sworn that main had to be in a public class in java < 25. You're right, I was wrong. This is indeed valid.
